cert/aziot-certd-config/src/util.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use crate::PreloadedCert;
+
+pub fn get_path(
+    homedir_path: &std::path::Path,
+    preloaded_certs: &std::collections::BTreeMap<String, PreloadedCert>,
+    cert_id: &str,
+) -> Result<std::path::PathBuf, Box<dyn std::error::Error + Send + Sync>> {
+    if let Some(preloaded_cert) = preloaded_certs.get(cert_id) {
+        let path = get_preloaded_cert_path(preloaded_cert, cert_id)?;
+        return Ok(path);
+    }
+
+    let mut path = homedir_path.to_owned();
+    path.push("certs");
+
+    if !path.exists() {
+        let () = std::fs::create_dir_all(&path)?;
```

Comment:
This should not be done when called from `aziot`. The permissions will be wrong.

Reply:
Wouldn't it just error due to not having permission to create the directory?

Reply:
(if that isn't the case, then I guess I can slap on a `create_dir: bool` parameter and leave it at that)

Reply:
`check` runs as root. You wouldn't be able to read the service configs otherwise.

Reply:
ah, very true. okay I feex